src/Hint/NewType.hs

```diff
@@ -138,3 +150,41 @@ dropConsBang x = x
 isUnboxedTuple :: HsType GhcPs -> Bool
 isUnboxedTuple (HsTupleTy _ HsUnboxedTuple _) = True
 isUnboxedTuple _ = False
+
+data WarnNewTypeUn = WarnNewTypeUn
```

Comment:
Why not a Maybe? Or even desugar the whole thing away, and have a warnNewTypeUn :: LHsDecl GhcPs -> String -> Idea`?